dns-server/src/dns_server.rs

```diff
@@ -287,118 +275,126 @@ async fn handle_dns_message(
     // have to decide if the error is authoritative.
     header.set_authoritative(true);
 
-    let query = mr.query();
-    let name = query.original().name().clone();
-    let answer = store.query(mr)?;
-    let rb = MessageResponseBuilder::from_message_request(mr);
     let mut additional_records = vec![];
 
-    let mut name_records = answer
-        .records
-        .as_ref()
-        .unwrap_or(&Vec::new())
-        .iter()
-        .map(|record| dns_record_to_record(&name, record))
-        .collect::<Result<Vec<_>, _>>()?;
-
-    if answer.name.is_none() && query.query_type() == RecordType::SOA {
-        // The query was for an SOA record at the apex. There isn't an SOA
-        // record in the database, but we can build one from the answer.
-        name_records.push(store.soa_for(&answer)?);
-    }
-
-    // If there were no records for the name at all, the name simply is not
-    // known to us. Bail now to return NXDomain.
-    //
-    // If there are no records after filtering, the name is known, just not with
-    // any records.  Returning NXDomain later on would be incorrect.
-    if name_records.is_empty() {
-        return Err(RequestError::NxDomain(answer.queried_fqdn()));
-    }
-
-    let response_records = name_records
+    let response_records = mr
```

Comment:
@iximeow this is the area where I would particularly appreciate scrutiny. hickory changed from a single query (`query()`) to multiple queries (`queries()`). I think I've updated this in a reasonable fashion, but I'm a bit out of my depth here.